apps/api/src/rest/routes/study-rooms.ts

```diff
@@ -0,0 +1,129 @@
+import { defaultHook } from "$hooks";
+import { productionCache } from "$middleware";
+import {
+  errorSchema,
+  responseSchema,
+  studyRoomSchema,
+  studyRoomsPathSchema,
+  studyRoomsQuerySchema,
+} from "$schema";
+import { StudyRoomsService } from "$services";
+import type { Bindings } from "$types/bindings";
+import { OpenAPIHono, createRoute } from "@hono/zod-openapi";
+import { database } from "@packages/db";
+
+const studyRoomsRouter = new OpenAPIHono<{ Bindings: Bindings }>({ defaultHook });
+
+const allStudyRoomsRoute = createRoute({
+  summary: "List all study rooms",
+  operationId: "allStudyRooms",
+  tags: ["Study Rooms"],
+  method: "get",
+  path: "/all",
+  description: "Retrieves all available study rooms.",
+  responses: {
+    200: {
+      content: {
+        "application/json": { schema: responseSchema(studyRoomSchema.array()) },
+      },
+      description: "Successful operation",
+    },
+    500: {
+      content: { "application/json": { schema: errorSchema } },
+      description: "Server error occurred",
+    },
+  },
+});
+
+const studyRoomByIdRoute = createRoute({
+  summary: "Retrieve a study room",
+  operationId: "studyRoomById",
+  tags: ["Study Rooms"],
+  method: "get",
+  path: "/{id}",
+  request: { params: studyRoomsPathSchema },
+  description: "Retrieves a study room by its ID.",
+  responses: {
+    200: {
+      content: { "application/json": { schema: responseSchema(studyRoomSchema) } },
+      description: "Successful operation",
+    },
+    404: {
+      content: { "application/json": { schema: errorSchema } },
+      description: "Study room not found",
+    },
+    422: {
+      content: { "application/json": { schema: errorSchema } },
+      description: "Parameters failed validation",
+    },
+    500: {
+      content: { "application/json": { schema: errorSchema } },
+      description: "Server error occurred",
+    },
+  },
+});
+
+const studyRoomsByFiltersRoute = createRoute({
+  summary: "Filter study rooms",
+  operationId: "studyRoomsByFilters",
+  tags: ["Study Rooms"],
+  method: "get",
+  path: "/",
+  request: { query: studyRoomsQuerySchema },
+  description: "Retrieves study rooms matching the given filters.",
+  responses: {
+    200: {
+      content: {
+        "application/json": { schema: responseSchema(studyRoomSchema.array()) },
+      },
+      description: "Successful operation",
+    },
+    422: {
+      content: { "application/json": { schema: errorSchema } },
+      description: "Parameters failed validation",
+    },
+    500: {
+      content: { "application/json": { schema: errorSchema } },
+      description: "Server error occurred",
+    },
+  },
+});
+
+studyRoomsRouter.get(
+  "*",
+  productionCache({ cacheName: "study-room-api", cacheControl: "max-age=86400" }),
+);
```

Comment:
We should probably not be caching this endpoint since it'll be providing real-ish time data of study room availability.